index.bs

```diff
@@ -577,66 +577,170 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
+		copying a selected piece of plain text from another app. In both cases, the MIME type in the pasteboard would be text/plain. This is used for both reading and writing.
+		</p>
+
+		<p>A [=clipboard item=] has an associated <dfn>types</dfn> which contains the MIME types.</p>
+
+		<dl class=note>
+			<dt><code><var>clipboardItem</var> = new ClipboardItem([<var>items</var>, <var>options</var>])</code>
+			<dd>
+			Creates a new {{ClipboardItem}} object. <var>items</var> are used to fill its MIME types and [=Promise=]s to [=Blob=]s or {{DOMString}}s corresponding to the MIME types, <var>options</var> can be used to fill its {{ClipboardItemOptions}},
+			as per the example below.
+
+				<pre class="example javascript" highlight=js>
+					const format1 = 'text/plain';
+					const promise_text_blob = Promise.resolve(new Blob(['hello'], {type: format1}));
+					const clipboardItemInput = new ClipboardItem(
+						{[format1]: promise_text_blob},
+						{options: "unspecified"});
+				</pre>
+
+			<dt><code><var>clipboardItem</var>.getType(<var>type</var>)</code>
+			<dd><p>Returns a [=Promise=] to the [=Blob=] corresponding to <var>type</var>.</p>
+
+			<dt><code><var>clipboardItem</var>.<var>types</var></code>
+			<dd><p>Returns the list of <var>types</var> contained in the <var>clipboardItem</var> object.
+
+		</dl>
+		<p>
+			A {{ClipboardItem}} object has an associated [=clipboard item=].
+		</p>
+		<p>
+			A {{ClipboardItem}} object's {{ClipboardItem/presentationStyle}} is its [=clipboard item=]'s [=presentation style=].
+		</p>
+		<p>
+			A {{ClipboardItem}} object's {{ClipboardItem/types}} is its [=clipboard item=]'s [=types=].
+		</p>
+		<p>
+			The <a constructor lt="ClipboardItem()">constructor</a> steps for <code>new ClipboardItem(<var>items</var>, <var>options</var>)</code> are to set [=this=]'s items to <var>items</var> and options to <var>options</var>.
+		</p>
+
+		<h4 attribute for=ClipboardItem lt=presentationStyle>presentationStyle</h4>
+		<p>
+			Each {{ClipboardItem}} has a {{ClipboardItem/presentationStyle}}, which is a {{PresentationStyle}}.
+			The {{ClipboardItem/presentationStyle}} getter steps are to return [=this=]'s [=clipboard item=]'s [=presentation style=].
+		</p>
+
+		<h4 attribute for=ClipboardItem lt=types>types</h4>
+		<p> Each {{ClipboardItem}} has a {{ClipboardItem/types}}, which contains the MIME types.
+		    {{ClipboardItem/types}} getter steps are to return [=this=]'s [=clipboard item=]'s [=types=].</p>
+
+		<h4 method for=ClipboardItem lt=getType(type)>getType(type)</a> must run the below steps:</h4>
+
+			1. Let |realm| be [=this=]'s [=relevant realm=].
+
+			1. If |type| is not listed in the [=mandatory data types=] list, then [=a promise rejected with=] "The type was not found" DOMException in |realm|.
+
+			1. Let |p| be [=a new promise=] in |realm|.
+
+			1. Let |blobData| be a [=Blob=] corresponding to the |type|.
```

Comment:
I have no idea how to implement this. My best guess is that if you passed in `text/plain` as a type, this would return a promise for the blob created with `new Blob(["text/plain"])`, i.e. containing the string "text/plain" in UTF-8 encoding.

Reply:
Clarified some steps using some new concepts defined in the `clipboard item` object.